src/pytest_inmanta_lsm/lsm_project.py

```diff
@@ -179,6 +187,43 @@ def lsm_services_update_attributes(
 
         return inmanta.protocol.common.Result(code=200, result={})
 
+    def lsm_services_update_attributes_v2(
+        self,
+        tid: uuid.UUID,
+        service_entity: str,
+        service_id: uuid.UUID,
+        current_version: int,
+        patch_id: str,
+        edit: typing.List[inmanta_lsm.model.PatchCallEdit],
+        comment: typing.Optional[str] = None,
+    ) -> inmanta.protocol.common.Result:
+        """
+        This is a mock for the lsm api, this method is called during allocation to update
+        the attributes of a V2 service.
+        """
+        # Making some basic checks
+        service = self.services[str(service_id)]
+        assert str(tid) == self.environment, f"{tid} != {self.environment}"
+        assert service.service_entity == service_entity, f"{service.service_entity} != {service_entity}"
+        assert service.version == current_version, f"{service.version} != {current_version}"
+
+        # The attributes parameter only represents the attributes that should be changed.
+        # * When no candidate attributes were set, the new candidate attributes will be equal to the active
+        #   attributes with the attribute updates applied.
+        # * When candidate attributes were set, the update will be applied to the existing candidate
+        #   attributes.
+        if service.candidate_attributes is None:
+            service.candidate_attributes = copy.deepcopy(service.active_attributes)
+
+        attrs = {}
+        for patch in edit:
+            attrs[patch.target] = patch.value
```

Comment:
This works for simple paths, for longer paths, this should use dictpath.

Reply:
You might be able to simply steal this logic here: https://github.com/inmanta/inmanta-lsm/blob/39e9319381ce6cfc9fd22549e2b5a9cc7128ded2/src/inmanta_lsm/model.py#L2794 (Calling the method with None instead of self should be fine, as this self parameter is not accessed there)

Reply:
Ah no, it does access self: https://github.com/inmanta/inmanta-lsm/blob/39e9319381ce6cfc9fd22549e2b5a9cc7128ded2/src/inmanta_lsm/model.py#L2836

So copying the logic above that line should be fine